### ISSUE TITLE
Analytics breaks intermittently with error `r[e] is undefined`

### ISSUE THREAD

**Issue by andfinally** [woothemes/woocommerce#33323]
### Describe the bug

On WooCommerce.com, the Analytics Overview page intermittently fails to render. When this happens, we get JS errors in the console.

### To Reproduce

1. Open devtools.
2. Go to the analytics page `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Foverview` on the WCCOM production or develop environments.

### Actual behavior:

The content of the page fails to render. On FF, the console shows this error:

```
 TypeError: r[e] is undefined
    M https://woocommerce.com/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=6.6.0:2
    promise callback*Gr< https://woocommerce.com/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=6.6.0:2
    f https://woocommerce.com/wp-includes/js/dist/vendor/react.min.js?ver=17.0.1:9
    os https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    Ur https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    Ir https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    Dr https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    Pr https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    Er https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    ul https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    ml https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    Lr https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    ml https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    render https://woocommerce.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=17.0.1:9
    <anonymous> https://woocommerce.com/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=6.6.0:2
    <anonymous> https://woocommerce.com/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=6.6.0:2
    <anonymous> https://woocommerce.com/wp-content/plugins/woocommerce/assets/client/admin/app/index.js?ver=6.6.0:2
```

On Chromium, we get a few errors. This one is more informative, because it points us to a `woocommerce_products_task_layout_card` ExPlat experiment:

```
index.js?ver=6.6.0:formatted:3515 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'loadExperimentAssignment')
    at i (index.js?ver=6.6.0:formatted:3515)
    at M (index.js?ver=6.6.0:formatted:3519)
    at index.js?ver=6.6.0:formatted:7115
    at Module.62969 (index.js?ver=6.6.0:formatted:7117)
    at M (index.js?ver=6.6.0:formatted:11792)
    at index.js?ver=6.6.0:formatted:12742
    at index.js?ver=6.6.0:formatted:17745
    at index.js?ver=6.6.0:formatted:17747
```

(If the page renders OK, do a hard refresh till you get the error.)

### Expected behavior

The analytics content loads as usual.

#### Chromium

<img width="662" alt="image" src="https://user-images.githubusercontent.com/1647564/172193214-89552bfa-fccd-450d-a9db-61cfeed9ca81.png">

### Desktop (please complete the following information):

* macOS
* FF 101.0 / Chromium 96.0

### Additional context

- This issue has been reported by WooCommerce.com vendors 13524-gh-Automattic/woocommerce.com.
- We are unable to reproduce it on our local WCCOM environments, but we see it on our production and develop environments.
- We thought this might be linked to the `_.contains is not a function` error we see around the same time. 
  - This error crops up in WordPress from time to time (13437-gh-Automattic/woocommerce.com) – it happens when a script with a lodash dependency which is not in `noConflict` mode overwrites the `_` global. Some wp-admin scripts rely on that global to contain underscore. 
  - But it seems not, because even when the analytics page doesn't render, `_` has the value `function cr(e)`, i.e. underscore, and even when it has the value `function lodash(value)` on our local environments, the page renders OK.
- We've tried updating WordPress to 6.0 and Jetpack to 10.9 to match production.
- We've tried reverting WooCommerce to 6.5.1.

### References

p6q8Tx-2hM-p2#comment-5970

**Comment by mehamasum** [comment]
@andfinally  I see a `Missing data from PHP (wpNotesArgs)` warning (on prod wccom, linux/chrome).
Just a hunch - some piece of data is missing from the backend and thus the client is failing to render?

![image](https://user-images.githubusercontent.com/12912469/172199960-a77cae65-e79b-4488-b2cc-9c65d8f7694f.png)

**Comment by chihsuan** [comment]
~Looks like analytics page is back to normal now?~

~I was able to view the page content though I'm still seeing some warnings/errors.~

Ah, I'm seeing the `r[e] is undefined` error again.

![Screen Shot 2022-06-07 at 10 37 14](https://user-images.githubusercontent.com/4344253/172283992-397acf5c-be21-480b-a3a6-8ba724fb2bd5.png)

**Comment by chihsuan** [comment]
@andfinally @mehamasum I think it's related to asset cache.

It fails when it tries to load a module function. And the module id generated by webpack is `85932`.

![Screen Shot 2022-06-07 at 12 13 04](https://user-images.githubusercontent.com/4344253/172294961-83b3a19f-8196-4c04-a9c2-5a2b2c05390d.png)

I searched the plugin files and found out it's `store-alert.js`

It seems that store-alert.js URL returns responses randomly. https://woocommerce.com/wp-content/plugins/woocommerce/assets/client/admin/chunks/store-alerts.js

Expected
![Screen Shot 2022-06-07 at 12 15 14](https://user-images.githubusercontent.com/4344253/172294974-3190991d-080b-4ea0-98f3-db836fc3952c.png)

Unexpected
![Screen Shot 2022-06-07 at 12 15 33](https://user-images.githubusercontent.com/4344253/172294979-88b910e0-1703-4df1-820e-bd182a661e0d.png)

**Comment by adrianduffell** [comment]
@chihsuan Does `store-alert.js` get requested with a version param? e.g. `store-alert.js?ver=6.6.0`

**Comment by chihsuan** [comment]
@adrianduffell Nope. The analytics page requests it without a version param. 

![Screen Shot 2022-06-07 at 12 39 53](https://user-images.githubusercontent.com/4344253/172297306-fda202cf-e004-412d-92f6-97be5f4b72a4.png)

Ah, that's the reason why we get random responses.

**Comment by adrianduffell** [comment]
> Ah, that's the reason why we get random responses.

I think so!

Ideally there would be a version param so a new cache key is generated with every new WooCommerce version.

**Comment by andfinally** [comment]
Thanks @chihsuan for that awesome insight! 🙌 And many thanks for your help @mehamasum and @adrianduffell.

I'm curious how you figured that out - I added a breakpoint at that line in `index.js`, but I never got that particular number. I'd also love to understand how to use the number to identify the file! But that is maybe a question for another day – the file now seems to be refreshed, and the page is rendering OK on develop and production. 🎉 🎉 🎉

**Comment by andfinally** [comment]
Our immediate issue was solved, but I'd better keep this issue open while https://github.com/woocommerce/woocommerce/pull/33332 is open.

**Comment by mattsherman** [comment]
#33332 is merged now, so closing this.

**Comment by chihsuan** [comment]
> I'm curious how you figured that out - I added a breakpoint at that line in index.js, but I never got that particular number. I'd also love to understand how to use the number to identify the file! But that is maybe a question for another day – the file now seems to be refreshed, and the page is rendering OK on develop and production.  

@andfinally Yeah, I'm so glad everything is working fine now. I'd be happy to share how I got the number. 😄 

Originally, I was also unable to get it with the breakpoint or using [pause on any exception](https://developer.chrome.com/blog/automatically-pause-on-any-exception/). I think it's because there are other exceptions, and the function is called many times.

So to pause on the exception at that line, I set a breakpoint on `r[e] === undefined` condition

<img src="https://user-images.githubusercontent.com/4344253/172827960-6cfcd4ac-c8d4-4b7d-973f-e51f192056e5.png"  width="650px" />

<img src="https://user-images.githubusercontent.com/4344253/172827630-fc14ca2c-eaa5-4585-bb01-6f8550f64bfd.png"  width="650px" />

And then, I noticed that `r` is an object that key is a number and value is a  module, so I thought `e` must be an id of exported module/function.

![Screen Shot 2022-06-09 at 18 48 23](https://user-images.githubusercontent.com/4344253/172829930-daed68c7-8706-4d41-b6ed-9ad6fdf50839.png)

Therefore, I just searched over all files in the woocommerce.com folder using VSCode and found that it's store-alert.js. 🙂
